### PR TITLE
[PLT-6838] Restrict channel delete option per permission policy even for last channel member

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Restrict channel delete option per permission policy even for last channel member

#### Ticket Link
Jira ticket: [PLT-6838](https://mattermost.atlassian.net/browse/PLT-6838)

#### Checklist
- [x] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by grundleborg** [comment]
Holding this one until the discussion at https://pre-release.mattermost.com/core/pl/tkfnniws77bptxngyg5scxw1ka has been concluded.

File: webapp/components/channel_header.jsx [hunk 1]
```diff
@@ -724,8 +724,6 @@ export default class ChannelHeader extends React.Component {
                 if (!ChannelStore.isDefault(channel)) {
                     dropdownContents.push(deleteOption);
                 }
-            } else if (this.state.userCount === 1) {
```
Comment (hmhealey) [inline webapp/components/channel_header.jsx]: This menu option should still exist for the last user of a private channel

**Comment by saturninoabril** [comment]
@hmhealey noted. see updated PR.

File: api4/channel.go [hunk 2]
```diff
@@ -540,17 +540,20 @@ func deleteChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Allow delete if user is the only member left in channel
-	if memberCount > 1 {
-		if channel.Type == model.CHANNEL_OPEN && !app.SessionHasPermissionToChannel(c.Session, channel.Id, model.PERMISSION_DELETE_PUBLIC_CHANNEL) {
-			c.SetPermissionError(model.PERMISSION_DELETE_PUBLIC_CHANNEL)
-			return
-		}
+	// Do not allow delete if there's only one member left in a private channel
```
Comment (grundleborg) [inline api4/channel.go]: This appears to stop the last user in a private channel from deleting it, even if they have the permissions to via team/channel admin policy. Is this intended?

Comment (saturninoabril) [inline api4/channel.go]: Yes, except when the user executing the delete is a system admin.

Comment (grundleborg) [inline api4/channel.go]: That doesn't make any sense to me. Let's assume that the `RestrictPrivateChannelDeletion` policy is set to `team` and I'm a `team_admin` but not a `system_admin`. This means that when the channel has 2 members I am allowed to delete it, but if I am the only member, I am not allowed to delete it any more.

**Pull request merged**